nasa/progpy
# 152 make tensor flow optional

This PR accomplishes 2 things
1. Makes installing tensor flow optional using [datadriven] option, and 
2. Updates configuration from old setup.py to new pyproject.toml

Now to install with tensorflow you must use the following command
pip install progpy[datadriven] or pip install '.[datadriven]' (for local copy)

Also updated LSTM to give a more descriptive error if tensor flow isn't installed and updated documentation of LSTM to add note.

How to test:
* Create a new venv
* Install progpy local copy (pip install -e .) , run tests. data tests should fail and give useful error.
* Clear out venv
* install progpy local copy, with datadriven (pip install -e '.[datadriven]'), run tests. all should pass
